director.go

```diff
@@ -24,6 +24,11 @@ func NewSceneDirector[T any](scene Scene[T], state T, RuleSet map[Scene[T]][]Dir
 
 // ProcessTrigger finds if a transition should be triggered
 func (d *SceneDirector[T]) ProcessTrigger(trigger SceneTransitionTrigger) {
+	if prevTransition, ok := d.current.(SceneTransition[T]); ok {
+		// previous transition is still running, end it to process trigger
+		prevTransition.End()
```

Comment:
There definitely still need to be a `return` right after this call to end, though I'm still not sure why only what I observe as the problem. I'll go ahead and push up my change to examples/director which adds a 3rd scene to highlight the issue a bit better than only 2 scenes.

Without `return` after `prevTransition.End()`: right clicking twice quickly through transitions brings you back to the first scene (Red), but if you wait for the transitions to finish it takes three right clicks

With `return` after `prevTransition.End()`: right clicking twice quickly through transitions brings you to the 3rd scene (Green) as expected, three right clicks needed to fully cycle back to Red when starting from Red.

Reply:
I think I found the problem. In fact, for that example `ProcessTrigger` is being called twice on the transition because both scenes are updating while transitioning. So the flow without the `return` is the following:
1. FirstScene is active and starts transition to ScondScene
2. While transitioning, the right click is pressed.
3. FirstScene calls `processTrigger`
4. The transition ends and SecondScene is now active.
5. The trigger is processed and start transition from SecondScene to the ThirdScene
6. SecondScene calls `processTrigger`
7. The transition ends and ThirdScene is now active
8. The trigger is processed and start transition from ThirdScene to FirstScene.

Adding the `return` works in this case because of this double call, the first cancels the transition and the second actually does the new transition.

Reply:
So is there some other change you're thinking of, or does adding the return sound like the way to go?

Reply:
Actually, I think that adding the `return` would lead to unexpected behavior. Let's think in this example:

The game is transitioning and the player press the pause button. We expect that the transition is cancelled and the pause menu(a third scene) is opened.

If we add the `return`, the player will go to the second scene and the pause menu will not open, since the pause trigger will not be processed. Without the `return` the menu will open as expected.

Checkout my updated example without the double trigger on the `fix-transition-panic` branch. I wasn't added to this pr for some reason 🥲

Reply:
Thanks for the explanation on how it's the processTrigger function is running from the updates taking input from both the from and to scenes, that and the [Transition Awareness documentation](https://github.com/joelschutz/stagehand?tab=readme-ov-file#transition-awareness) help illustrate this usage.

I've sync'd the fork to take the updated example, however it seems to add some unnecessary complication to the example. Someone who might want to use it as a simple example or reference implementation may not as easily follow what it's actually trying to do with the "Condition" variable.

There's also a point where I was able to left click to get the count on the first scene (red) above the "condition" max count and then get stuck on the red screen unable to ever get the "can switch" back to true without restarting the application.


All that said I think I can accept the explanation on the double run of processTrigger due to the from/to scene update functions, I think it might be best to go back to a simpler director example and then the rest should be ok.

Reply:
I might have a simpler implementation for the example that is both straight forward and prevents this double calling. Checkout the branch, I'll be adding it very soon.

Reply:
I just syncd the fork with your change, seems perfect now!